### ISSUE TITLE
DecoderException in HexaCord -> PaperSpigot 1.7.10 + ViaVersion

### ISSUE THREAD

**Issue by MrPowerGamerBR** [HexagonMC/BungeeCord#96]
```
15:39:32 [SEVERE] [player_test] <-> ServerConnector [lobby] - encountered exception
io.netty.handler.codec.DecoderException: java.lang.IndexOutOfBoundsException: readerIndex(6) + length(14080) exceeds writerIndex(14): UnpooledSlicedByteBuf(ridx: 6, widx: 14, cap: 14/14, unwrapped: PooledUnsafeDirectByteBuf(ridx: 15, widx: 15, cap: 1024))
    at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:98)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:350)
    at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:267)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:350)
    at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:266)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:350)
    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
    at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926)
    at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:972)
    at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:386)
    at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:302)
    at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IndexOutOfBoundsException: readerIndex(6) + length(14080) exceeds writerIndex(14): UnpooledSlicedByteBuf(ridx: 6, widx: 14, cap: 14/14, unwrapped: PooledUnsafeDirectByteBuf(ridx: 15, widx: 15, cap: 1024))
    at io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1395)
    at io.netty.buffer.AbstractByteBuf.checkReadableBytes(AbstractByteBuf.java:1389)
    at io.netty.buffer.AbstractByteBuf.readBytes(AbstractByteBuf.java:850)
    at io.netty.buffer.AbstractByteBuf.readBytes(AbstractByteBuf.java:858)
    at net.md_5.bungee.protocol.DefinedPacket.readArrayLegacy(DefinedPacket.java:103)
    at net.md_5.bungee.protocol.packet.EncryptionRequest.read(EncryptionRequest.java:30)
    at net.md_5.bungee.protocol.MinecraftDecoder.decode(MinecraftDecoder.java:33)
    at net.md_5.bungee.protocol.MinecraftDecoder.decode(MinecraftDecoder.java:10)
    at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:88)
    ... 21 more
```

Okay, I bit of context: I use a private ViaVersion fork which works in 1.7.10, so my server allows 1.7 - 1.10.2 connections.

I can connect to the server just fine without HexaCord, however if I add ViaVersion on my server and I try to join via HexaCord that error is thrown in the console.

It is strange, because that error only happens *if* I try to join via HexaCord, but I can connect just fine without BungeeCord.

I also can share my private ViaVersion fork + my private PaperSpigot 1.7.10 fork (because it is needed because PaperSpigot 1.7.10 has outdated Guava libraries) if needed.

**Comment by i9hdkill** [comment]
As 1.7.10 is working expected with vanilla spigot, I guess your custom ViaVersion is messing stuff up

**Comment by MrPowerGamerBR** [comment]
@i9hdkill yeah, finding the issue will be a bit difficult (since... well, my "custom ViaVersion" is just the latest ViaVersion in GitHub but I replaced every "io.netty" with "net.minecraft.util.io.netty")

Strange that it does work without any issues if I don't use HexaCord, but if I use it HexaCord it throws that error in the console (and only in HexaCord, the server just shows that the client disconnected)

**Comment by i9hdkill** [comment]
Did you try 1.7-1.8 BungeeCord?

**Comment by MrPowerGamerBR** [comment]
@i9hdkill can you share a 1.7 - 1.8 BungeeCord build so I can test it? Thanks!

**Comment by i9hdkill** [comment]
http://ci.md-5.net/job/BungeeCord/1119/

**Comment by MrPowerGamerBR** [comment]
@i9hdkill yep, same error.

Welp, I think I will need to find out what is causing this issue by intercepting the packets, strange that it only happens when using ViaVersion (even tho ViaVersion shouldn't affect the 1.7.X protocol since I made it to detect that the server is a 1.8 (47) server, not a 1.7 server)

**Comment by MrPowerGamerBR** [comment]
@i9hdkill I'm facepalming so hard right now.

I forgot that @Matsv made ViaVersion to always tries to process the packets, even if it is a 1.8 client, this was because ViaVersion BossBar API was having some issues.

Well, I forgot to revert that commit, and guess what happens when you don't forget to remove that commit? It changes the ENCRYPTION_BEGIN packet if you are using 1.7.X (because, as I said before, ViaVersion tries to process it as it was a 1.8.X packet) and HexaCord says "oh shit what the fuck is that" and throws an error.

Reverting the commit fixes the issue in 1.7.X clients, so, yeah, it wasn't HexaCord fault (and neither ViaVersion fault, it was my fault for forgetting about that commit until now)

![http://i.imgur.com/H24ZzKP.png](http://i.imgur.com/H24ZzKP.png)

**Comment by i9hdkill** [comment]
Oh, good to know, thx for coming back to us to explain the "issue":)
